test/lib/testMayaUsdLayerEditorCommands.py

```diff
@@ -604,6 +604,31 @@ def testLayerLockWritePermission(self):
         # 6- Checking that at least one layer is modifiable
         self.assertTrue(mayaUsdUfe.isAnyLayerModifiable(stage))
     
+    def testRefreshSystemLockCallback(self):
+        # FileBacked Layer Write Permission
+        # 1- Loading the test scene
+        rootLayerPath = testUtils.getTestScene("layerLocking", "layerLocking.usda")
+        stage = Usd.Stage.Open(rootLayerPath)
+        topLayer = stage.GetRootLayer();
+        layerLockingShapes = cmds.ls(type="mayaUsdProxyShapeBase", long=True)
+        proxyShapePath = layerLockingShapes[0]
+        # 2- Setting a system lock on a layer loaded from a file
+        cmds.mayaUsdLayerEditor(topLayer.identifier, edit=True, lockLayer=(2, 0, proxyShapePath))
+        self.assertFalse(topLayer.permissionToEdit)
+        self.assertFalse(topLayer.permissionToSave)
+        # 3- Attach callbacks to capture any system-lock changes due to refreshSystemLock
+        def refreshSystemLockCallback(context, callbackData):
+            layerIds = callbackData.get('affectedLayerIds')
+            for layerId in layerIds:
+                mel.eval(f'mayaUsdLayerEditor -edit -lockLayer 2 0 "{proxyShapePath}" "{layerId}"')
+        
+        from usdUfe import registerUICallback
+        mayaUsd.lib.registerUICallback('onRefreshSystemLock', refreshSystemLockCallback)        
+        # 4- Refreshing the system lock should remove the lock but also get re-locked by refreshSystemLockCallback
+        cmds.mayaUsdLayerEditor(topLayer.identifier, edit=True, refreshSystemLock=(proxyShapePath, 1))
```

Comment:
One issue I can see is that the layer are system-locked, so if this call fails, they won't be refreshed, but the test will pass because the lack of refresh means they will still be locked.

Maybe instead of re-locking, the callback could capture the list of layer IDs it receive and assert that you have receive the expected layer ID instead of their lock state? (Might also then assert that they did get unlocked?)

Reply:
Good call, I'll update the test to check for the ids.